plugins/module_utils/user.py

```diff
@@ -161,14 +184,26 @@ def user_add(cursor, user, host, host_all, password, encrypted,
 
     mogrify = do_not_mogrify_requires if old_user_mgmt else mogrify_requires
 
+    # This is for update_password: on_new_username
     used_existing_password = False
     if reuse_existing_password:
-        existing_auth = get_existing_authentication(cursor, user, host)
+        existing_auth = get_existing_authentication(cursor, user)
         if existing_auth:
-            plugin = existing_auth['plugin']
-            plugin_hash_string = existing_auth['plugin_hash_string']
-            password = None
-            used_existing_password = True
+            pass_hashes = [p['plugin_hash_string'] for p in existing_auth]
+            # Use a set to check if all values are the same
+            if len(set(pass_hashes)) != 1:
+                module.warn("An account with the username %s has a different "
+                            "password than the others existing accounts. Thus "
+                            "on_new_username can't decide which password to "
+                            "reuse so it will use your provided password "
+                            "instead. If no password is provided, the account "
+                            "will have an empty password!" % user)
+                used_existing_password = False
+            else:
+                plugin_hash_string = existing_auth[0]['plugin_hash_string']
+                password = None
+                used_existing_password = True
+            plugin = existing_auth[0]['plugin']  # What if plugin differ?
```

Comment:
I would say the line above should be indented by one level.
Taking plugin_hash_string from an existing user only makes sense when also taking the same plugin.
When taking the plugin_hash_string or password from the module input, the plugin from module input should be used.

Reply:
I believe the indentation is fine.
`existing_auth` contains 2 columns:
- plugin_hash_string
- plugin
The if..else takes care of the first one
Line 206 the second one.

Do you agree @betanummeric?

Reply:
@laurent-indermuehle Sorry, no. `plugin_hash_string` and `plugin` only make sense together. User authentication won't work with `plugin_hash_string` in plugin-A-style and `plugin` set to plugin B.

## proposed behavior
When `update_password: on_new_username` and
- no matching username was found
- or multiple matching usernames were found, but also varying plugin_hash_string/plugin

take `plugin`, `password`, `plugin_hash_string` etc. all from the module input and nothing from the pre-existing users.

It was like this before. We can get this by indenting the line. Please do this.

## currently in this PR
When `update_password: on_new_username` and multiple matching usernames were found, but also varying plugin_hash_string/plugin,
take a random `plugin` of those found and all other values from module input. This may be a different plugin than what the `plugin_hash_string` from the module arguments was made for.

This may work, but there is no guarantee. It changes behavior. I would consider this a bug.

Reply:
Thanks @betanummeric, I added the indentation.
I hope the missing 'plugin` variable won't cause an undefined value down the line. I expect the CI to catch that (we will see in 20 min).

Reply:
Mmh, It broke tests for MySQL 8 only... Have to debug more :(

Reply:
@betanummeric I've found the issue. It was a test I added following our conversation. And because the default plugin is different since MySQL 8+ it is normal that test failed. It is fixed now. Ready for a hopefully last review please.